src/impulse.c

```diff
@@ -58,15 +59,18 @@ static void quit( int ret ) {
 
 static void get_source_info_callback( pa_context *c, const pa_source_info *i, int is_last, void *userdata ) {
 
-	if ( is_last || device != NULL )
+	if(swept_through)
 		return;
+	if ( is_last || device != NULL )
+		swept_through = 1;
+
+	printf("%d->%s\n",i->index, i->name);
 
 	assert(i);
```

Comment:
I have the same problem with both `0` and `1`
This is the line that crashes, I have managed to fix it by changing `if(swept_through)` with `if(swept_through || !i)`
I will open a PR shortly, if you think there is a better way to solve it, please tell me